src/OpenFeature.Contrib.Providers.Flagd/Resolver/InProcess/CustomEvaluators/FractionalEvaluator.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JsonLogic.Net;
+using Murmur;
+using Newtonsoft.Json.Linq;
+using Semver;
+
+namespace OpenFeature.Contrib.Providers.Flagd.Resolver.InProcess.CustomEvaluators
+{
+    /// <inheritdoc/>
+    public class FractionalEvaluator
+    {
+
+        class FractionalEvaluationDistribution
+        {
+            public string variant;
+            public int percentage;
+        }
+
+        internal object Evaluate(IProcessJsonLogic p, JToken[] args, object data)
+        {
+            // check if we have at least two arguments:
+            // 1. the property value
+            // 2. the array containing the buckets
+
+            if (args.Length < 2)
+            {
+                return null;
+            }
```

Comment:
There's a shorthand version uses the `targetingKey` if the bucketing expression is missing, see the flowchart [here](https://flagd.dev/reference/specifications/custom-operations/fractional-operation-spec/).

Please add a test for this as well!

Reply:
thanks, seems like i missed that fallback behaviour - i have adapted the implementation to use the targetingKey now :)